src/test/java/hashtable/HashTableTest.java

```diff
@@ -0,0 +1,178 @@
+package hashtable;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class HashTableTest {
+
+    @Test(expected = IllegalArgumentException.class)
+    public void impossibleToSetNegativeCapacity() {
+        new HashTable(-1);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void impossibleToSetZeroCapacity() {
+        new HashTable(0);
+    }
+
+    @Test
+    public void possibleToSetPositiveCapacity() {
+        new HashTable(2);
+    }
+
+    @Test
+    public void possibleToPutElementInEmptySlot() {
```

Comment:
We shouldn't test private parts of the algorithm. We have public API like `getValue()`, `add()`, `remove()` - these are the methods that we can use for testing.